### PR TITLE
Skip auth check in case of localhost usage

### LINKED ISSUE #11461: Allow usage of WMCore HTTP REST server in local environment
**Impact of the new feature**
As developer we need to test changes to REST HTTP server in our local setup, i.e. without apache. This will help to avoid spending time on setup of either dedicated VM or usage of k8s where apache frontend which performs authentication should be present. In dev environment, e.g. using local laptop, we do not need this setup if we only care about features or development needed for REST HTTP server.

**Is your feature request related to a problem? Please describe.**
So far REST HTTP server checks presence of CMS HTTP headers which are provided by CMSWEB frontends. Therefore, in local environment, e.g. laptop, we need additional effort to provide these headers, either by setting up apache FE, or fixing (manually) the codebase to skip authentication.

**Describe the solution you'd like**
In developer environment we should be able to skip authentication when using localhost.

**Describe alternatives you've considered**
I need to modify auth function in REST server when using local environment.

**Additional context**

### PR DESCRIPTION
**Pull request by vkuznet**
Fixes #11461 

#### Status
ready

#### Description
Provide simple fix to check HTTP headers and skip authentication if localhost is used.

#### Is it backward compatible (if not, which system it affects?)
YES

### CODE REVIEW

**Review by amaltaro — CHANGES_REQUESTED** [review]
It looks like unit tests need to be updated as well.

**Comment by vkuznet** [comment]
@amaltaro , I fixed `Simple_t` unit test by changing status, but I added appropriate comment. Please see in a code. The rest of failure in unit test are due to unstable tests.

File: src/python/WMCore/REST/Auth.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def user_info_from_headers(key, verbose=False):
 
     # Reject if request was not authenticated.
     if 'cms-auth-status' not in headers:
+        if headers.get('Remote-Addr', '') == '127.0.0.1':
+            # since we are using localhost access we can skip authentication
```
Comment (amaltaro) [inline src/python/WMCore/REST/Auth.py:29]: Valentin, I would suggest to take the same approach as the block below. So, please add a log here to identify that unauthenticated access has been performed (e.g.):
```
log("DEBUG: non-authenticated request performed")
```

**Review by amaltaro — COMMENTED** [review]
Valentin, I have made just a minor suggestion in the code. Thanks

File: src/python/WMCore/REST/Auth.py (hunk 1 above)
Comment (vkuznet) [reply src/python/WMCore/REST/Auth.py:29]: done

**Review by amaltaro — APPROVED** [review]

**Comment by amaltaro** [comment]
@mapellidario @belforte this might be handy as well when debugging CRABServer from localhost. Just FYI

**Comment by belforte** [comment]
thanks Alan. But AFAICT we only had to worry about authentication in the old days when running CMSWEB FE and CRABServer on a VM. Now we do all work in K8s pods. Maybe I miss some other improvement which you guys did which makes testing CMSWEB apps even easier ?

**Comment by amaltaro** [comment]
Valentin (and Todor) have been running specific WMCore services in their own node/laptop, with this change in, there is no need to have frontends performing the authentication part and adding extra HTTP request headers to the http call.

I haven't tried this myself though as I didn't try to run any of the central services in my own laptop :)

**Comment by vkuznet** [comment]
Stefano, the changes I made will allow to run, debug code in local environment, eg laptop, without requiring FE presence.

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by belforte** [comment]
I see. Thanks Valentin and Alan !

**Pull request merged**